### PR TITLE
Add support for data page statistics

### PR DESCRIPTION
**Pull request by akrennmair**
This PR adds support for data page statistics. To do this, I moved the recording of minimum and maximum values to a separate set of types, and thus added the ability to distinguish between min/max per row group and min/max per data page. I also added the correct number of rows contained in the data page.

### CODE REVIEW

**Comment by panamafrancis** [comment]
Why do dictionary stores not need min/max statistics?

File: type_bytearray.go [hunk 1]
```diff
@@ -292,12 +292,21 @@ func (b *byteArrayDeltaEncoder) Close() error {
 }
 
 type byteArrayStore struct {
-	repTyp   parquet.FieldRepetitionType
-	min, max []byte
+	repTyp parquet.FieldRepetitionType
+	st     stats
+	pagest stats
```
Comment (panamafrancis) [inline type_bytearray.go:297]: pageSt? or stPage? or call 'stats' 'statistics' instead and the struct fields stats & pageStats?

**Comment by panamafrancis** [comment]
Where are the stats being tested?

File: type_bytearray.go (hunk 1 above)
Comment (akrennmair) [reply type_bytearray.go]: I renamed several identifiers to make things clearer.

**Comment by akrennmair** [comment]
@panamafrancis regarding your questions:

`dictStore` only fulfills the role to collect values and collect information about null values as well as distinct values. It does not operate in the same function as `typedColumnStore` which is embedded by `ColumnStore` which is also a user of `dictStore`. This area definitely require a refactoring as the purpose of `dictStore` has been trimmed down to only a minimum, and some of the naming is certainly outdated and possibly confusing.

As for testing, this only generates  data page statistics on the write side, and there exists no equivalent support on the read side. So we have no built-in way of introspecting data page statistics for testing purposes. What I did for testing was [parquet-cli](https://github.com/apache/parquet-mr/tree/master/parquet-cli) which allows introspection with the `pages` command, and also has its own command `check-stats` that does a check whether all the statistics, both on row group and page level, are actually correct. Using that tool, I checked all parquet files generated by the tests, and none showed any issues, [unlike before](https://github.com/fraugster/parquet-go/issues/69), btw.

**Review by panamafrancis — APPROVED** [review]

**Pull request merged**